examples/step-81/step-81.cc

```diff
@@ -0,0 +1,799 @@
+    void make_grid();
+    void setup_system();
+    void assemble_system();
+    void solve();
+    void output_results();
+
+    Parameters<dim>            parameters;
+    PerfectlyMatchedLayer<dim> perfectly_matched_layer;
+
+    Triangulation<dim> triangulation;
+    DoFHandler<dim>    dof_handler;
+
+    std::unique_ptr<FiniteElement<dim>> fe;
+
+    AffineConstraints<double> constraints;
+    SparsityPattern           sparsity_pattern;
+    SparseMatrix<double>      system_matrix;
+    Vector<double>            solution;
+    Vector<double>            system_rhs;
+  };
+
+
+  // @sect4{The Constructor}
+  // The Constructor simply consists specifications for the mesh
+  // and the order of the fnite elements. These are editable through
+  // the .prm file.
+
+  template <int dim>
+  Maxwell<dim>::Maxwell()
+    : ParameterAcceptor("Maxwell")
+    , dof_handler(triangulation)
+  {
+    ParameterAcceptor::parse_parameters_call_back.connect(
+      std::bind(&Maxwell<dim>::parse_parameters_callback, this));
+
+    scaling = 20;
+    add_parameter("scaling", scaling, "scale of the hypercube geometry");
+
+    refinements = 6;
+    add_parameter("refinements",
+                  refinements,
+                  "number of refinements of the geometry");
+
+    fe_order = 0;
+    add_parameter("fe order", fe_order, "order of the finite element space");
+
+    quadrature_order = 1;
+    add_parameter("quadrature order",
+                  quadrature_order,
+                  "order of the quadrature");
+  }
+
+
+  template <int dim>
+  void Maxwell<dim>::parse_parameters_callback()
+  {
+    fe = std::make_unique<FESystem<dim>>(FE_NedelecSZ<dim>(fe_order), 2);
+  }
+
+  // Make the mesh for the domain and generate the triangulation required.
+  // Additionally, there is an interface added here to visualize
+  // a standing wave. To generate a solution without any interface,
+  // comment out lines 455-459.
+
+  template <int dim>
+  void Maxwell<dim>::make_grid()
+  {
+    GridGenerator::hyper_cube(triangulation, -scaling, scaling);
+    triangulation.refine_global(refinements);
+
+    for (auto &cell : triangulation.active_cell_iterators())
+      if (cell->center()[1] > 0.)
+        cell->set_material_id(1);
+      else
+        cell->set_material_id(2);
+
+    std::cout << "Number of active cells: " << triangulation.n_active_cells()
+              << std::endl;
+  }
+ 
+  // Enumerate all the degrees of freedom and set up matrix and vector
+  // objects to hold the system data. Enumerating is done by using
+  // DoFHandler::distribute_dofs().
+
+  template <int dim>
+  void Maxwell<dim>::setup_system()
+  {
+    dof_handler.distribute_dofs(*fe);
+    std::cout << "Number of degrees of freedom: " << dof_handler.n_dofs()
+              << std::endl;
+
+    solution.reinit(dof_handler.n_dofs());
+    system_rhs.reinit(dof_handler.n_dofs());
+    constraints.clear();
+    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
```

Comment:
At least in the moment, the FE_NedelecSZ class does not have support for hanging nodes.
Therefore `DoFTools::make_hanging_node_constraints` should not be called at this point.

That there is no warning when calling `DoFTools::make_hanging_node_constraints` has already been mentioned in Issue #12775.